js/index.js

```diff
@@ -38,5 +38,25 @@ export default class Teamshares {
       },
       mutator: svg => svg.setAttribute("fill", "currentColor"),
     });
+
+    // Register FontAwesome Pro and point at our bespoke "kit" URL, including the token in the querystring
+    // See https://fontawesome.com/kits/44da2a9d09/setup
+    // Note that the kit allows us to whitelist / deny specific URL patterns
+    Shoelace.registerIconLibrary("fa", {
+      resolver: name => {
+        const filename = name.replace(/^fa([rsltdb]|(ss))-/, "");
+        const sub = name.substring(0, 4);
+        const folderHash = {
+          "fas-": "solid",
+          "fal-": "light",
+          "fat-": "thin",
+          "fad-": "duotone",
+          "fab-": "brands",
+        };
+        const folder = folderHash[sub] || "regular";
+        return `https://ka-p.fontawesome.com/releases/v6.4.0/svgs/${folder}/${filename}.svg?token=44da2a9d09`;
```

Comment:
The token is included in all the URL requests, so it doesn't seem like much of a secret. That being the case, I didn't see a problem with hard-coding it here.